src/database/Database.cpp

```diff
@@ -293,6 +294,50 @@ Database::dropTxMetaIfExists()
     }
 }
 
+void
+Database::maybeUpgradeToBucketListDB()
+{
+    if (mApp.getPersistentState().getState(PersistentState::kDBBackend) !=
+        BucketIndex::DB_BACKEND_STATE)
+    {
+        CLOG_INFO(Database, "Upgrading to BucketListDB");
+
+        // Drop all LedgerEntry tables except for offers
+        CLOG_INFO(Database, "Dropping table accounts");
+        getSession() << "DROP TABLE IF EXISTS accounts;";
+
+        CLOG_INFO(Database, "Dropping table signers");
+        getSession() << "DROP TABLE IF EXISTS signers;";
+
+        CLOG_INFO(Database, "Dropping table claimablebalance");
+        getSession() << "DROP TABLE IF EXISTS claimablebalance;";
+
+        CLOG_INFO(Database, "Dropping table configsettings");
+        getSession() << "DROP TABLE IF EXISTS configsettings;";
+
+        CLOG_INFO(Database, "Dropping table contractcode");
+        getSession() << "DROP TABLE IF EXISTS contractcode;";
+
+        CLOG_INFO(Database, "Dropping table contractdata");
+        getSession() << "DROP TABLE IF EXISTS contractdata;";
+
+        CLOG_INFO(Database, "Dropping table accountdata");
+        getSession() << "DROP TABLE IF EXISTS accountdata;";
+
+        CLOG_INFO(Database, "Dropping table liquiditypool");
+        getSession() << "DROP TABLE IF EXISTS liquiditypool;";
+
+        CLOG_INFO(Database, "Dropping table trustlines");
+        getSession() << "DROP TABLE IF EXISTS trustlines;";
+
+        CLOG_INFO(Database, "Dropping table ttl");
+        getSession() << "DROP TABLE IF EXISTS ttl;";
+
+        mApp.getPersistentState().setState(PersistentState::kDBBackend,
```

Comment:
Once nodes do this "migration", does BucketListDB just work or do operators need to run some command like new-db? Or I guess it can work right away, just need to stall for a bit to rebuild the indexes, right?

Reply:
It depends. If the `run` user has DB admin privileges, it will just work. If not, the operator will have to call `upgrade-db` with the correct user, so it pretty much works like a DB schema upgrade.